R/smtp_send.R

```diff
@@ -19,57 +32,71 @@
 #'   carbon copies. Any email addresses provided here will receive the message
 #'   and these email addresses will be concealed from other recipients
 #'   (including others on the BCC list).
-#' @param creds_file An optional path to an email credentials file. This file
-#'   must be created by the `create_email_creds_file()` function.
-#' @param sender The sender name.
-#' @param host The email host.
-#' @param port The port associated with the email account.
-#' @param user The username associated with the email account.
-#' @param password The password associated with the email account.
-#' @param use_ssl A logical value to indicate whether to use SSL.
-#' @param authenticate A logical value to indicate whether to use
-#'   authentication.
+#' @param credentials One of three credential helper functions must be used
+#'   here: (1) [creds_file()], (2) [creds_key()], or (3) [creds()]. The first,
+#'   [creds_file()], relies on a credentials file stored on disk. Such a file is
+#'   created using the [create_smtp_creds_file()] function. The [creds_key()]
+#'   function is used if credentials are stored in the system-wide key-value
+#'   store, through use of the [create_smtp_creds_key()] function. Using
+#'   [creds()] allows for a manual specification of SMTP configuration and
+#'   credentials within that helper function.
```

Comment:
Encourage people to use `creds` if possible--it's most secure because the password is simply not stored. `creds_key` is second most secure but it's not possible to move the file around. `creds_file` is useful when you need to deploy a project somewhere else and have the credentials move too--already not ideal from a security perspective but sometimes necessary.

Reply:
I promoted `creds()` to the top of the list and also explicitly stated that it's the most secure means of providing credentials (all within the internal documentation). I also do this elsewhere (to reinforce this notion).